src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -0,0 +1,20 @@
+import EnrollIdPBody from './EnrollAuthenticatorIdPBody';
+import ChallengeIdpBody from './ChallengeAuthenticatorIdPBody';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    authenticatorEnrollments.filter(authenticator => authenticator.id === currentAuthenticator.id);
+
+    // Set the view based on whether or not we're in a challenge flow
+    this.Body = authenticatorEnrollments.length
+      ? ChallengeIdpBody
+      : EnrollIdPBody;
+  },
+  Footer: AuthenticatorEnrollFooter,
```

Comment:
We have an `AuthenticatorEnrollFooter` and an `AuthenticatorVerifyFooter`, so you might want to do the same thing you are doing with `Body`